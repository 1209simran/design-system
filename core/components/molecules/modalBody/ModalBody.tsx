import * as React from 'react';
import classNames from 'classnames';

const { useRef, useEffect, useState } = React;

interface IModalBodyProps {
  children: React.ReactNode;
}

const ModalBody: React.FunctionComponent<IModalBodyProps> = props => {
  const [scroll, setScroll] = useState(false);
  const ref = useRef<HTMLDivElement>(null);
  const { children } = props;

  useEffect(() => {
    const scrollHeight = ref && ref.current ? ref.current.scrollHeight : 0;
    const clientHeight = ref && ref.current ? ref.current.clientHeight : 0;
    if (scrollHeight > clientHeight) {
      setScroll(true);
    }
  }, [ref]);

  const classes = classNames({
    'Modal-body': true,
    ['Modal-body--border']: scroll
  });

  return (
    <div className={classes} ref={ref}>
      {children}
    </div>
  );
};

export default ModalBody;
