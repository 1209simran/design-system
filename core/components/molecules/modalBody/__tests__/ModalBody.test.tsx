import * as React from 'react';
import { shallow } from 'enzyme';
import ModalBody from '../ModalBody';

describe('ModalBody component', () => {
  it('should match the snapshot for Modal Body without scroll', () => {
    const tree = shallow(
      <ModalBody>
        <p>Modal Body</p>
      </ModalBody>
    );

    expect(tree).toMatchSnapshot();
  });

  it('should match the snapshot for Modal Body with scroll', () => {
    const tree = shallow(
      <ModalBody>
        <div style={{ height: '500px' }}>
          <p>Modal Body</p>
        </div>
      </ModalBody>
    );

    expect(tree).toMatchSnapshot();
  });
});
