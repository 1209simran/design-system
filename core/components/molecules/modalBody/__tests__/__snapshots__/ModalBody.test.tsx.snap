// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBody component should match the snapshot for Modal Body with scroll 1`] = `
<div
  className="Modal-body"
>
  <div
    style={
      Object {
        "height": "500px",
      }
    }
  >
    <p>
      Modal Body
    </p>
  </div>
</div>
`;

exports[`ModalBody component should match the snapshot for Modal Body without scroll 1`] = `
<div
  className="Modal-body"
>
  <p>
    Modal Body
  </p>
</div>
`;
